src/replication.c

```diff
@@ -750,6 +766,225 @@ is_replica_newly_connected(spec_t *spec, replica_t *new_replica)
 	return newly_connected;
 }
 
+// get_known_replica_count return no.of known replicas
+static int
+get_known_replica_count(spec_t *spec) {
+	int known_replica_count = 0;
+	trusty_replica_t *knr;
+	TAILQ_FOREACH(knr, &spec->lu->known_replica, next) {
+		known_replica_count++;
+	}
+	return known_replica_count;
+}
+
+//TODO: Instead of connect each time use epoll to make connection
+// and remove hardcoded values
+// update_volume_config must be performed by taking spec lock
+static int
+update_volume_config(char *data) {
+#ifdef DEBUG
+	return 1;
+	ISTGT_LOG("successfully updated the volume into etcd\n");
+#else
+	int fd, rc = -1, len;
+	int rtimeout = 20, wtimeout = 10, tmpidx = 0, tmpcnt = 0;
+	char *read_data, *tmp_data;
+	const char err[] = "Err\n";
+
+	fd = istgt_connect_unx(ISTGT_MGMT_UNXPATH);
+	if (fd < 0) {
+		ISTGT_ERRLOG("failed to connect to %s\n", ISTGT_MGMT_UNXPATH);
+		goto error_in_update;
+	}
+	len = strlen(data);
+
+       /* set timeout msec. */
+        rc = istgt_set_recvtimeout(fd, rtimeout * 1000);
+        if (rc != 0) {
+                ISTGT_ERRLOG("istgt_set_recvtimeo() failed\n");
+		rc = -1;
+                goto error_in_update;
+        }
+        rc = istgt_set_sendtimeout(fd, wtimeout * 1000);
+        if (rc != 0) {
+                ISTGT_ERRLOG("istgt_set_sendtimeo() failed\n");
+		rc = -1;
+                goto error_in_update;
+        }
+
+	rc = istgt_writeline_socket(fd, data, wtimeout);
+	if (rc < 0) {
+		ISTGT_ERRLOG("failed to write the data on %d no_of_bytes written %d\n", fd, rc);
+		goto error_in_update;
+	} else if (rc < len) {
+		ISTGT_ERRLOG("partial data written on %d no_of_bytes written %d\n", fd, rc);
+		rc = -1;
+		goto error_in_update;
+	}
+	//TODO: Improve read code
+	read_data = (char *)malloc(sizeof(char) * 128);
+	tmp_data = (char *)malloc(sizeof(char) * 64);
+	rc = istgt_readline_socket(fd, read_data, 128, tmp_data, 64, &tmpidx, &tmpcnt, rtimeout);
+	if (rc < 0) {
+		ISTGT_ERRLOG("failed to read the data on %d\n", fd);
+		goto clean_data;
+	}
+	if(strcmp(read_data, err) == 0 || strlen(read_data) == 0) {
+		ISTGT_ERRLOG("failed to update %s\n", read_data);
+		rc = -1;
+		goto clean_data;
+	}
+	rc = 1;
+clean_data:
+	free(read_data);
+	free(tmp_data);
+error_in_update:
+	close(fd);
```

Comment:
this can get called even if fd < 0.. need to have the check

Reply:
updated the code